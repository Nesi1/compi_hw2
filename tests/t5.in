int main()
{
    int a;
    if (a)
        if (c)
            if (d)
                d1=2;
            else
                d2=3;
        else
            c = 2; 
}