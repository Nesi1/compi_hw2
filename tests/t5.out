31: Type -> INT
5: RetType -> Type
7: Formals -> epsilon
31: Type -> INT
15: Statement -> Type ID SC
12: Statements -> Statement
36: Exp -> ID
36: Exp -> ID
36: Exp -> ID
38: Exp -> NUM
18: Statement -> ID ASSIGN Exp SC
38: Exp -> NUM
18: Statement -> ID ASSIGN Exp SC
23: Statement -> IF LPAREN Exp RPAREN Statement ELSE Statement
38: Exp -> NUM
18: Statement -> ID ASSIGN Exp SC
23: Statement -> IF LPAREN Exp RPAREN Statement ELSE Statement
22: Statement -> IF LPAREN Exp RPAREN Statement
13: Statements -> Statements Statement
4: FuncDecl -> RetType ID LPAREN Formals RPAREN LBRACE Statements RBRACE
2: Funcs -> epsilon
3: Funcs -> FuncDecl Funcs
1: Program -> Funcs
